### PR TITLE
attach to neovim on windows

### LINKED ISSUE #157: cannot connect on windows
Hi guys, i can run the qt client on windows fine.

I'm trying to connect using the python client now.

I set the NVIM_LISTEN_ADDRESS variable but i receive an error
```python
vim = neovim.attach('socket', path=os.environ['NVIM_LISTEN_ADDRESS'])
```
```
File "C:\Python34\lib\asyncio\events.py", line 330, in create_unix_connection
raise NotImplementedError
```
I've read that others can get it working.. is this a new bug or am i missing a step?

### PR DESCRIPTION
**Pull request by timeyyy**
updated _connect socket function and tests to work with windows.

fixes #157

### CODE REVIEW

File: neovim/msgpack_rpc/event_loop/asyncio.py [hunk 1]
```diff
@@ -111,6 +114,11 @@ def _threadsafe_call(self, fn):
         self._loop.call_soon_threadsafe(fn)
 
     def _setup_signals(self, signals):
+        if os.name == 'nt':
+            # add_signal_handler is not supported in win32
+            self._signals = []
+            return
+
```
Comment (equalsraf) [inline neovim/msgpack_rpc/event_loop/asyncio.py]: There are some trailing whitespaces in this line.

**Comment by brcolow** [comment]
You should add @equalsraf 's name as the original author of the patch IMHO (credit where credit is due). He may not care, but he deserves it :)

**Comment by timeyyy** [comment]
@brclow do you mean in the commit comment or something else?

**Comment by brcolow** [comment]
In the commit comment =)

**Comment by timeyyy** [comment]
white space fixed and credit given.

**Comment by timeyyy** [comment]
One test is still failing (which was there previously). I think a new issue should be made for it.

**Comment by bfredl** [comment]
I made #161 for the linter failure.

File: neovim/msgpack_rpc/event_loop/asyncio.py [hunk 2]
```diff
@@ -111,6 +114,11 @@ def _threadsafe_call(self, fn):
         self._loop.call_soon_threadsafe(fn)
 
     def _setup_signals(self, signals):
+        if os.name == 'nt':
+            # add_signal_handler is not supported in win32
+            self._signals = []
+            return
```
Comment (justinmk) [inline neovim/msgpack_rpc/event_loop/asyncio.py]: shouldn't this be an error instead of silently failing?

**Comment by justinmk** [comment]
is `os.name == 'nt'` the typical way in python to check for windows platform?

**Comment by timeyyy** [comment]
@justinmk It is standard.

**Comment by timeyyy** [comment]
@equalsraf is the brains behind the patch, in the issue it was mentioned that weneed more input in regards to the signals

File: neovim/msgpack_rpc/event_loop/asyncio.py (hunk 2 above)
Comment (equalsraf) [inline neovim/msgpack_rpc/event_loop/asyncio.py]: Currently signal support is not working - https://github.com/neovim/python-client/issues/157#issuecomment-183994226 and https://github.com/neovim/python-client/issues/157#issuecomment-183997724 - maybe add a TODO: here?

**Comment by bfredl** [comment]
Is this RDY? I haven't tested it, but if it enables attaching on windows let's merge it.

**Comment by maxwell-k** [comment]
I'm testing this at the moment; I can get most nose tests to pass, but I still get an error when I try to use  `nvim --cmd "python3 print('test')"`:


```
2016-02-23 20:14:32,808 [ERROR @ base_events.py:default_exception_handler:1008] 8000 - Fatal read error on pipe transport
protocol: <neovim.msgpack_rpc.event_loop.asyncio.AsyncioEventLoop object at 0x008ADAD0>
transport: <_ProactorReadPipeTransport fd=0>
Traceback (most recent call last):
  File "C:\Users\maxwellk\AppData\Local\Programs\Python34\lib\asyncio\proactor_events.py", line 191, in _loop_reading
    self._read_fut = self._loop._proactor.recv(self._sock, 4096)
  File "C:\Users\maxwellk\AppData\Local\Programs\Python34\lib\asyncio\windows_events.py", line 425, in recv
    self._register_with_iocp(conn)
  File "C:\Users\maxwellk\AppData\Local\Programs\Python34\lib\asyncio\windows_events.py", line 606, in _register_with_iocp
    _overlapped.CreateIoCompletionPort(obj.fileno(), self._iocp, 0, 0)
OSError: [WinError 6] The handle is invalid
```

**Comment by timeyyy** [comment]
@maxwell-k can you confirm that from master you cannot attach to neovim, while after applying this patch you can?

**Comment by maxwell-k** [comment]
@timeyyy With master I can't connect at all. With this patch I can connect from a Python 3 REPL or the nose tests to a running `nvim-qt` but `:python3 print('test')` within `nvim-qt` produces the error above

**Comment by bfredl** [comment]
but this PR is for attach not starting python from within nvim right?

**Comment by timeyyy** [comment]
@bfredl correct, so we can now do `from neovim import attach`  and use it successfully. 

We still have a failing test on windows but as mentioned in the pull request that will be posted as a new issue. 

@maxwell-k thanks for confirming

**Comment by bfredl** [comment]
So, a step forward on windows, but no change of behavior on unix. Merged.

**Pull request merged**